src/execute/index.js

```diff
@@ -261,16 +307,14 @@ export function buildRequest(options) {
       throw new Error(`Required parameter ${parameter.name} is not provided`);
     }
 
-    if (
-      specIsOAS3 &&
-      parameter.schema &&
-      parameter.schema.type === 'object' &&
-      typeof value === 'string'
-    ) {
-      try {
-        value = JSON.parse(value);
-      } catch (e) {
-        throw new Error('Could not parse object parameter value string as JSON');
+    if (specIsOAS3 && typeof value === 'string') {
+      if (findObjectSchema(parameter.schema, { recurse: false })) {
+        value = parseJsonValue({ value, shouldFailSafe: false });
+      } else if (
+        isNil(parameter.schema?.type) &&
```

Comment:
NITPICK: can we use native JS constructs if they can be expressed in simple way? In this particular case:

```suggestion
        parameter.schema?.type == null &&
```

We can further refine logic here and skip the execution of recursive search when type is defined with whatever value. User can be define it as `type: null` or `type: undefined` and we SHOULD skip the recursive traversal in this case as well. We cannot use `Object.hasOwn()` as this library still supports Node.js >= 12.20.0. But we can use `R.has` from ramda.